### PR TITLE
PDAL scala

### PR DESCRIPTION
**Pull request by pomadchin**
PDAL Scala support

Adds Pipeline DSL with some hand written types for JSON generation.
DSL Coverage is not complete, for that purposes was introduced a special type which allows to write in a raw JSON.

- [x] Pipeline DSL
- [x] ~Pipeline DSL Generation~
- [x] ~Exclude generated files from index~
- [x] ~More Scala code on top of all jni types~
- [x] Travis CI
- [x] Snapshot
- [ ] Publishment 
  - [x] 1.5.0-RC1
  - [x] 1.5.0-RC2
  - [x] 1.5.0-RC3
  - [ ] 1.5.0
  - [ ] 1.6.0

### CODE REVIEW

**Comment by hobu** [comment]
What are .idea files?

**Comment by pomadchin** [comment]
@hobu garbage, i'll create a clean PR, without bad history.

**Comment by pomadchin** [comment]
@hobu I removed this folder from this branch, thanks!

**Comment by pomadchin** [comment]
The way right now generator works: parsing json output of `pdal --options all --showjson` command. To make cleaner generation that would be great to add `type` explicitly, and `requirement: {true | false}`, as right now it's unclear what fields are optional. 

Example is LAS Reader where only `filename` is a required field.

File: scala/.gitignore [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Test data files #
+
+java/data
+
+# Compiled libs #
+
+java/*.dylib
+java/*.so
+java/*dll
```
Comment (lossyrob) [inline scala/.gitignore]: Should the `pdal-scala` subproject just be part of the `java` directory?

Comment (pomadchin) [inline scala/.gitignore]: It's the same .gitignore; i think it makes sense to move it into java directory, as a subproject of already existence structure.

File: scala/project/Commands.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+object Commands {
+  def processJavastyleCommand(commandProcess: String) = {
+    Command.command(s"${commandProcess}-javastyle")((state: State) => {
+      val extracted = Project extract state
+      import extracted._
+      val publishState = Command.process(commandProcess, append(Seq(crossPaths := false), state))
+      append(Seq(crossPaths := true), publishState)
+    })
+  }
+}
```
Comment (lossyrob) [inline scala/project/Commands.scala]: What's this?

**Comment by hobu** [comment]
Should this be merged or closed?

**Comment by pomadchin** [comment]
It requires some small refactors, and would be ready for a final review. I'll try to push this task soon.

File: java/core-scala/src/main/scala/io/pdal/pipeline/ReaderTypes.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+  * Copyright (c) 2017, hobu Inc.  (info@hobu.co)
```
Comment (lossyrob) [inline java/core-scala/src/main/scala/io/pdal/pipeline/ReaderTypes.scala]: I think the copyright on these Scala bits should be to Azavea. Happy to sign a CLA to allow for things like future relicensing.

File: java/README.md [hunk 4]
```diff
@@ -28,6 +28,20 @@ It's required to have native JNI binary in `java.library.path`:
...
+
+```scala
+libraryDependencies ++= Seq(
+  "io.pdal" %% "pdal-scala" % "1.5.0"
+)
+```
+
+Scala API covers PDAL 1.5.0 but is compatible with PDAL >= 1.4, to use any custom DSL
+that is not covered by the current Scala API you can use `RawExpr` type to build `Pipeline 
+Expression`.
```
Comment (lossyrob) [inline java/README.md]: Can we get some usage examples here? E.g. one without RawExpr and one with?

**Pull request merged**